### PR TITLE
Fix #2694: Remove deprecated methods from KubernetesClient DSL

### LINKED ISSUE #2694: Remove deprecated methods
Kubernetes Client 5.0 already breaks many things, so why not do a cleanup for all the deprecated methods?

We are talking about a MAJOR version upgrade, so it should be "safe" to break API compatibility here and have a clean API.

Some examples include:
```
KubernetesClient.customResourceDefinitions();
KubernetesClient.customResources(CustomResourceDefinition crd, Class<T> resourceType, Class<L> listClass);
KubernetesClient.customResource(CustomResourceDefinition crd, Class<T> resourceType, Class<L> listClass);
KubernetesClient.events();
...
(and all others)
```

This has the potential risk that people migrating to this version would not know what it's the alternative, but the risk could be reduced by writing the items in the migration guide.

Probably this would be undesirable, but there are few major upgrades that allow being this harsh. But if the intention is to be more polite, then please close this now.

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #2694 
Removing the following deprecated methods from KubernetesClient DSL:
- `client.customResourceDefinitions()`
- `client.events()`
- `client.customResources(CustomResourceDefinitionContext crd, Class<T>
  resourceType, Class<L> listClass)`
- `client.customResources(CustomResourceDefinition crd, Class<T>
  resourceType, Class<L> listClass)`
- `client.customResource(CustomResourceDefinition crd, Class<T>
  resourceType, Class<L> listClass)`

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/KubernetesClient.java [hunk 1]
```diff
@@ -140,78 +123,6 @@
...
-   *
-   * @param crdContext CustomResourceDefinitionContext describes the core fields used to search for CustomResources
-   * @param resourceType Class for CustomResource
-   * @param listClass Class for list object for CustomResource
-   * @param <T> T type represents CustomResource type. If it's Namespaced resource, it must implement
-   *           io.fabric8.kubernetes.api.model.Namespaced
-   * @param <L> L type represents CustomResourceList type
-   * @return returns a MixedOperation object with which you can do basic CustomResource operations
-   */
-  <T extends HasMetadata, L extends KubernetesResourceList<T>> MixedOperation<T, L, Resource<T>> customResources(CustomResourceDefinitionContext crdContext, Class<T> resourceType, Class<L> listClass);
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/KubernetesClient.java:164]: I don't see any deprecation annotation or JavaDoc tag for this method. Why did you remove it?

If it's not an error, the procedure would be to annotate it now with the `@Depreacted` annotation and remove it in future releases.

Comment (rohanKanojia) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/KubernetesClient.java:164]: These methods don't work as we need to specify annotations in POJOs https://github.com/fabric8io/kubernetes-client/issues/2738#issuecomment-763869356 . I don't think it's a good idea to keep methods that don't work at all.

Comment (manusa) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/KubernetesClient.java:164]: Fair enough, but that's out of scope of the current issue>PR.

Ideally you should create a separate PR for this, or at least put it in a different commit.

Comment (rohanKanojia) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/KubernetesClient.java:164]: okay, I'll remove this from this commit and create a separate PR in next update

**Review by manusa — APPROVED** [review]

**Pull request merged**